app/controllers/attributes/names_controller.rb

```diff
@@ -0,0 +1,17 @@
+class Attributes::NamesController < AttributesController
+  def show
```

Comment:
I really like this return to restful routes. So +1 for the `:show`

I'm now tinking about how far we might take the descriptive private methods:

```ruby
  def show
  
    render json: {
      govuk_account_session: @govuk_account_session.serialise,
      values:  get_attribute_values_from_remote_data_store,
    }
  rescue OidcClient::OAuthFailure
    head :unauthorized
  end

private

  def get_attribute_values_from_remote_data_store
      remote_attributes = get_attributes_from_params(
        params.fetch(:attributes),
        permission_level: :check,
      )

    values = @govuk_account_session.get_remote_attributes(remote_attributes)
    values.compact.keys
  end
  ```

Is it any clearer if we moved some of this into private methods?

Reply:
That `show` wouldn't be correct, because calling `get_remote_attributes` can change the result of `serialise`.  To protect against silent failure, it'll instead throw an error:
- https://github.com/alphagov/account-api/blob/main/app/lib/account_session.rb#L60
- https://github.com/alphagov/account-api/blob/main/app/lib/account_session.rb#L96

Reply:
We had a similar discussion [here](https://github.com/alphagov/account-api/pull/49#discussion_r620193764). Maybe that's a sign that this isn't very clear and we should improve it. Probably not in scope for this PR, but is it possible to make getting attributes not change the result of `serialise`? 🤔

Reply:
We could store the OAuth access and refresh token in the account-api database, rather than the session, but then we have to decide when to expire them (or the database will grow indefinitely) - currently session expiration is entirely a client-side thing.

Reply:
We've got a potential work-around in slack